src/makeCalculator.js

```diff
@@ -37,7 +37,41 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  let calculator = {};
+
+  calculator = {
+    add: (value) => {
+      return calculator.result + value;
+    },
+
+    subtract: (value) => {
+      return calculator.result - value;
+    },
+
+    multiply: (value) => {
+      return calculator.result * value;
+    },
+
+    divide: (value) => {
+      return calculator.result / value;
+    },
+
+    result: 0,
+
+    reset: () => {
+      calculator.result = 0;
+
+      return calculator;
+    },
+
+    operate: (operation, value) => {
+      calculator.result = operation(value);
+
+      return calculator;
+    },
```

Comment:
You can use this on operate and reset
When you use constructors, classes, or something like that this looks to service which call your method
calculator.reset() -> this on reset is a calculator
But on run callback, the function doesn’t have this:
add() -> this isn’t exit here